piptools/_compat/path_compat.py

```diff
@@ -0,0 +1,36 @@
+"""
+Compatibility helpers for working with paths and ``pathlib`` across platforms
+and Python versions.
+"""
+
+from __future__ import annotations
+
+import os.path
+import pathlib
+import sys
+
+
+def relative_to_walk_up(path: pathlib.Path, start: pathlib.Path) -> pathlib.Path:
```

Comment:
Let's be explicit about what's expected to be importable
```suggestion
__all__ = ("relative_to_walk_up",)


def relative_to_walk_up(path: pathlib.Path, start: pathlib.Path) -> pathlib.Path:
```